### PR TITLE
fix(vulnerability/lerna): update to fix security vulnerability

### PR DESCRIPTION
**Pull request by JAdshead**

### CODE REVIEW

File: packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js [hunk 1]
```diff
@@ -19,6 +19,8 @@
 const fs = require('fs');
 const path = require('path');
 const webpack = require('webpack');
+// disabling this eslint warning as this is being reported erroneously as a dependency
+// eslint-disable-next-line import/no-extraneous-dependencies
 const merge = require('webpack-merge');
```
Comment (nellyk) [inline packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js]: disabled eslint for this as this was needed for test and i don't think it was necessary for it be installed as a dependency

File: package.json [hunk 2]
```diff
@@ -15,21 +15,32 @@
...
   "devDependencies": {
+    "@babel/cli": "^7.2.3",
     "@babel/core": "^7.2.2",
     "@babel/polyfill": "^7.2.5",
     "@commitlint/cli": "^8.2.0",
     "@commitlint/config-conventional": "^8.2.0",
-    "amex-jest-preset-react": "^5.0.1",
-    "enzyme": "^3.9.0",
-    "enzyme-to-json": "^3.2.2",
+    "amex-jest-preset-react": "^4.0.0",
```
Comment (10xLaCroixDrinker) [inline package.json]: why was this downgraded?

File: package.json [hunk 3]
```diff
@@ -15,21 +15,32 @@
...
+    "babel-loader": "^8.0.5",
+    "babel-plugin-add-module-exports": "^0.2.1",
+    "babel-preset-amex": "^2.0.0",
+    "enzyme": "^3.1.0",
+    "enzyme-to-json": "^3.1.2",
     "eslint": "^4.12.1",
     "eslint-config-amex": "^7.1.0",
+    "holocron": "^1.0.0",
```
Comment (10xLaCroixDrinker) [inline package.json]: Does this really belong here?

File: package.json (hunk 2 above)
Comment (nellyk) [reply package.json]: @10xLaCroixDrinker Thank you, upgrading this

Comment (nellyk) [reply package.json]: 9b1abbb updated

File: package.json (hunk 3 above)
Comment (nellyk) [reply package.json]: removed this here 9b1abbb

**Comment by JAdshead** [comment]
@nellyk Thanks for updating this

File: packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js (hunk 1 above)
Comment (anescobar1991) [reply packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js]: hmm this is odd... I think a better explanation should be added in the comment for why you are disabling it. `being reported erroneously` is too generic and I dont fully understand why it is not a needed dependency.

This might only be working by coincidence of having another dependency bringing `webpack-merge` in so the right thing to do may be to add it as a dev dep.

File: package.json [hunk 4]
```diff
@@ -15,21 +15,30 @@
...
-    "amex-jest-preset-react": "^5.0.1",
-    "enzyme": "^3.9.0",
-    "enzyme-to-json": "^3.2.2",
-    "eslint": "^4.12.1",
+    "amex-jest-preset-react": "^5.0.2",
+    "babel-loader": "^8.0.5",
+    "babel-plugin-add-module-exports": "^0.2.1",
+    "babel-preset-amex": "^3.3.0",
+    "enzyme": "^3.11.0",
+    "enzyme-to-json": "^3.1.2",
```
Comment (mtomcal) [inline package.json]: Looks like this was downgraded as well.

File: package.json [hunk 5]
```diff
@@ -15,21 +15,30 @@
...
     "jest": "^24.1.0",
-    "lerna": "^2.9.0",
+    "lerna": "^3.19.0",
+    "nodemon": "^1.10.2",
     "react": "^16.11.0",
-    "react-dom": "^16.11.0",
-    "react-test-renderer": "^16.11.0"
+    "react-addons-test-utils": "^15.4.2",
+    "react-dom": "^16.0.0",
+    "react-test-renderer": "^16.0.0",
```
Comment (mtomcal) [inline package.json]: Perhaps keep these at `^16.11.0`

File: packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js (hunk 1 above)
Comment (nellyk) [reply packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js]: sound good, i'll add this `webpack-merge` back. Thanks for the feedback

File: package.json (hunk 4 above)
Comment (nellyk) [reply package.json]: Thank you catching this @mtomcal

File: packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js (hunk 1 above)
Comment (nellyk) [reply packages/holocron-module-register-webpack-plugin/__tests__/HolocronModuleRegisterPlugin.spec.js]: updated this on f88f66e

File: package.json (hunk 4 above)
Comment (nellyk) [reply package.json]: updated this dependencies on f88f66e

File: package.json (hunk 5 above)
Comment (nellyk) [reply package.json]: updated this on f88f66e

File: packages/holocron/__tests__/holocronModule.spec.jsx [hunk 6]
```diff
@@ -340,14 +365,28 @@ describe('holocronModule', () => {
...
+    })(({ moduleState: { x }, y, xy }) => (
+      <div>
+        {x}
+        {' '}
+*
+        {' '}
+        {y}
+        {' '}
+=
+        {' '}
```
Comment (10xLaCroixDrinker) [inline packages/holocron/__tests__/holocronModule.spec.jsx]: we really need to change this rule. @nellyk can you open a PR to the eslint config when you have a chance?

File: packages/holocron/src/constants.js [hunk 7]
```diff
@@ -14,15 +14,11 @@
...
-
-export {
-  REGISTER_MODULE_REDUCER,
-  MODULE_LOADED,
-  MODULE_LOAD_FAILED,
-  MODULE_LOADING,
-  MODULE_REDUCER_ADDED,
-  INIT_MODULE_STATE,
-} from './ducks/load';
+export const REGISTER_MODULE_REDUCER = '@@holocron/REGISTER_MODULE_REDUCER';
```
Comment (10xLaCroixDrinker) [inline packages/holocron/src/constants.js]: why did these move?

File: packages/holocron/src/ducks/load.js [hunk 8]
```diff
@@ -14,14 +14,10 @@
 
 import { Map as iMap, Set as iSet } from 'immutable';
 import { getModule, getModuleMap } from '../moduleRegistry';
-import { HOLOCRON_STORE_KEY, MODULES_STORE_KEY, REDUCER_KEY } from '../constants';
-
-export const REGISTER_MODULE_REDUCER = '@@holocron/REGISTER_MODULE_REDUCER';
-export const MODULE_LOADED = '@@holocron/MODULE_LOADED';
-export const MODULE_LOAD_FAILED = '@@holocron/MODULE_LOAD_FAILED';
-export const MODULE_LOADING = '@@holocron/MODULE_LOADING';
-export const MODULE_REDUCER_ADDED = '@@holocron/MODULE_REDUCER_ADDED';
```
Comment (10xLaCroixDrinker) [inline packages/holocron/src/ducks/load.js]: taking out the action types breaks the ducks pattern

File: packages/holocron/src/holocronModule.jsx [hunk 9]
```diff
@@ -55,11 +55,16 @@ export default function holocronModule({
...
+      // This needs to be removed to support React17
+      // eslint-disable-next-line camelcase
+      UNSAFE_componentWillReceiveProps(nextProps) {
+        const { loadCount } = this.state;
+        const { props } = this;
+        if (shouldModuleReload && shouldModuleReload(props, nextProps)) {
+          const newLoadCount = loadCount + 1;
           this.setState({ status: 'loading', loadCount: newLoadCount });
-          this.initiateLoad(newLoadCount, nextProps);
+          this.initiateLoad(newLoadCount, nextProps, loadCount);
```
Comment (10xLaCroixDrinker) [inline packages/holocron/src/holocronModule.jsx]: this looks like an unrelated change. why was it done?

File: package.json [hunk 10]
```diff
@@ -15,21 +15,24 @@
...
-    "@commitlint/config-conventional": "^8.2.0",
-    "amex-jest-preset-react": "^5.0.1",
-    "enzyme": "^3.9.0",
+    "@babel/cli": "^7.8.3",
+    "@babel/core": "^7.8.3",
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-conventional": "^8.3.4",
+    "amex-jest-preset-react": "^5.0.2",
+    "babel-preset-amex": "^2.1.0",
+    "core-js": "^3.6.4",
```
Comment (PixnBits) [inline package.json]: is core-js a direct dependency or is it brought in via jest or maybe amex-jest-preset-react?

Comment (PixnBits) [reply package.json]: 🤦‍♂️sorry, was thinking about jsdom due to some slack messages
core-js is babel related

File: packages/holocron-module-route/src/ModuleRouteUtils.jsx [hunk 11]
```diff
@@ -17,22 +17,22 @@ import { createRoutes } from '@americanexpress/one-app-router';
...
-  return Array.isArray(routes) ? routes.map(route => addToRouteProps(route, newProps)) :
-    [addToRouteProps(routes, newProps)];
+  return Array.isArray(routes) ? routes.map((route) => addToRouteProps(route, newProps))
+    : [addToRouteProps(routes, newProps)];
 };
 
-export const getRouteIndex = (routes, props) =>
-  (routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute);
+export const getRouteIndex = (
+  routes, props) => (routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute);
```
[the remark this answers is not in the document]
Comment (PixnBits) [reply packages/holocron-module-route/src/ModuleRouteUtils.jsx]:
```
export const getRouteIndex = (routes, props) => (
  routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute
);
```
or
```
export const getRouteIndex = (
  routes, props
) => (routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute);
```
or
```
export const getRouteIndex = (
  routes, props
) => (
  routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute
);
```
seem more readable to me but that's my opinion 😅
I'm curious what the rule was that pushed for this change 🙏

File: packages/holocron-module-route/src/ModuleRouteUtils.jsx [hunk 12]
```diff
@@ -97,7 +97,8 @@ export function createModuleRouteFromElement({ type, props }) {
     const children = passChildrenProps(route.children, { store });
     const childRoutes = createRoutesFromReactChildren(children, route);
 
-    if (childRoutes.length) {
+    // eslint-disable-next-line unicorn/explicit-length-check
+    if (childRoutes.length > 0) {
```
[the remark this answers is not in the document]
Comment (PixnBits) [reply packages/holocron-module-route/src/ModuleRouteUtils.jsx]: `>= 1` might be more appropriate than _just_ 0? 🤙

File: packages/holocron/__tests__/ducks/load.spec.js [hunk 13]
```diff
@@ -13,15 +13,16 @@
  */
 
 import { fromJS, Set as iSet, Map as iMap } from 'immutable';
-import { REDUCER_KEY } from '../../src/constants';
-import { getModuleMap } from '../../src/moduleRegistry';
-
-import reducer, {
-  REGISTER_MODULE_REDUCER,
+import {
+  REDUCER_KEY, REGISTER_MODULE_REDUCER,
```
Comment (PixnBits) [inline packages/holocron/__tests__/ducks/load.spec.js]: line break for consistency

File: packages/holocron/__tests__/holocronModule.spec.jsx [hunk 14]
```diff
@@ -116,8 +116,13 @@ describe('holocronModule', () => {
     const Module = holocronModule({
       name: 'mock-module',
       load,
-    })(({ moduleLoadStatus }) => <div>Mock Module - {moduleLoadStatus}</div>);
-    const mockStore = createStore(state => state);
+    })(({ moduleLoadStatus }) => (
+      <div>
+Mock Module -
```
Comment (PixnBits) [inline packages/holocron/__tests__/holocronModule.spec.jsx]: odd intentation, can this be pushed over? I'm guessing it affected a snapshot/test passing?

Comment (PixnBits) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: similar for other instances in this file

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 6 above)
Comment (PixnBits) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: is it related to [indentation](https://github.com/americanexpress/holocron/pull/3/files#r371485456)? (is my comment a duplicate? 😅)

File: packages/holocron/__tests__/holocronModule.spec.jsx [hunk 15]
```diff
@@ -340,14 +365,28 @@ describe('holocronModule', () => {
     const reducer = () => fromJS({ x: 3 });
     const mergeProps = (stateProps, dispatchProps, ownProps) => {
       const xy = stateProps.moduleState.x * ownProps.y;
-      return Object.assign({}, stateProps, dispatchProps, ownProps, { xy });
+      return {
+        ...stateProps, ...dispatchProps, ...ownProps, xy,
```
Comment (PixnBits) [inline packages/holocron/__tests__/holocronModule.spec.jsx]: @10xLaCroixDrinker probably a #nitpick 😓but in your opinion would this be more readable on separate lines? i.e.:
```
return {
  ...stateProps,
  ...dispatchProps,
  ...ownProps, 
  xy,
};
```
the `xy` kinda sneaks in at the end on one line, might be more readily visible which portions are spread and which are not (added as direct properties)?

File: packages/holocron/__tests__/loadModule.web.spec.js [hunk 16]
```diff
@@ -16,15 +16,21 @@ import { fromJS } from 'immutable';
 import loadModule from '../src/loadModule.web';
 import { resetModuleRegistry } from '../src/moduleRegistry';
 
+const map = {};
```
Comment (PixnBits) [inline packages/holocron/__tests__/loadModule.web.spec.js]: might be better to name the variable by what it's a map of? looks like a bag of registered mock event listeners, maybe `mockElementListeners`?

Comment (PixnBits) [reply packages/holocron/__tests__/loadModule.web.spec.js]: looking again, it appears listeners are called via `mockElement.addEventListener.mock.calls[0][1](..)`, is this `map` var needed? (maybe left over from debugging?)

File: packages/holocron/__tests__/loadModule.web.spec.js [hunk 17]
```diff
@@ -143,11 +149,11 @@ describe('loadModule.web', () => {
       })
     );
     const loadError = new Error('failed to load module');
-    mockElement.onerror(loadError);
-    await expect(loadPromise).rejects.toBe(loadError);
+    expect(mockElement.addEventListener.mock.calls[0][1](loadError)).toBeUndefined();
```
Comment (PixnBits) [inline packages/holocron/__tests__/loadModule.web.spec.js]: might want to assert that this is the listener for the call to `addEventListener` with `'error'`?

File: packages/holocron/src/loadModule.web.js [hunk 18]
```diff
@@ -50,16 +50,15 @@ export default function loadModule(moduleName, moduleData) {
     }
     script.src = isProduction ? `${url}?key=${getModuleMap().get('key')}` : url;
     const timeout = setTimeout(onScriptComplete, 120000);
-    script.onerror = (err) => {
-      script.onerror = null;
+    script.addEventListener('error', (event) => {
```
Comment (PixnBits) [inline packages/holocron/src/loadModule.web.js]: was this change from `onerror` to `addEventListener('error'` tested in browsers? ex: IE11

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 15 above)
Comment (PixnBits) [inline packages/holocron/__tests__/holocronModule.spec.jsx]: trying something new, by suggestion of @JAdshead & @mtomcal so 🤞
```suggestion
        ...stateProps,
        ...dispatchProps,
        ...ownProps, 
        xy,
```

Comment (PixnBits) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: trying this out with a code suggestion in https://github.com/americanexpress/holocron/pull/3/files#r371494636

File: packages/holocron/src/RenderModule.jsx [hunk 19]
```diff
@@ -29,6 +29,8 @@ export default function RenderModule({ children, moduleName, props }) {
 
   const propsWithChildren = typeof children !== 'undefined' ? { children, ...props } : props;
 
+  // disabling this because propsWithChildren is computed and passed down to this prop
```
Comment (PixnBits) [inline packages/holocron/src/RenderModule.jsx]: maybe
```suggestion
  // propsWithChildren is computed and passed down to this prop
```
for brevity? #nitpick

File: packages/holocron/__tests__/createHolocronStore.spec.js [hunk 20]
```diff
@@ -47,12 +47,12 @@ describe('createHolocronStore', () => {
   const defaultAppState = 'app state';
   const defaultAppReducer = (state = { app: defaultAppState }) => state;
   const helpers = (function helpers() {
-    const getStateFromSource = cb => ({ store, state } = {}) =>
-      cb(state || store ? store.getState() : iMap());
+    const getStateFromSource = (cb) => (
+      { store, state } = {}) => cb(state || store ? store.getState() : iMap());
```
[the remark this answers is not in the document]
Comment (PixnBits) [reply packages/holocron/__tests__/createHolocronStore.spec.js]: there are quite a few uncommented
@nellyk are you running prettier locally on save by chance? 🤔

File: packages/holocron-module-route/src/ModuleRouteUtils.jsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (nellyk) [reply packages/holocron-module-route/src/ModuleRouteUtils.jsx]: happy use either i'll remove the comment

File: packages/holocron/__tests__/createHolocronStore.spec.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (nellyk) [reply packages/holocron/__tests__/createHolocronStore.spec.js]: nope, let me review the rest

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 6 above)
Comment (nellyk) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: ooh yeah agreed

File: packages/holocron/__tests__/__snapshots__/holocronModule.spec.jsx.snap [hunk 21]
```diff
@@ -2,38 +2,42 @@
...
   loading
 </div>
 `;
 
 exports[`holocronModule should pass the moduleLoadStatus prop as loaded when loaded 1`] = `
 <div>
-  Mock Module - 
-  loaded
+  Mock Module -
+  loading
```
Comment (nellyk) [inline packages/holocron/__tests__/__snapshots__/holocronModule.spec.jsx.snap]: need to fix this too as this should be loaded

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 6 above)
Comment (nellyk) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: https://github.com/americanexpress/eslint-config-amex/pull/8

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 15 above)
Comment (nellyk) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: I'm leaning towards the first one as you can see all the props in one line

File: packages/holocron/__tests__/loadModule.web.spec.js (hunk 16 above)
Comment (nellyk) [reply packages/holocron/__tests__/loadModule.web.spec.js]: using it on line `31` to populate the list of events listeners to be added to the mock. Let me know if  that is what you meant?

File: packages/holocron/src/holocronModule.jsx (hunk 9 above)
Comment (nellyk) [reply packages/holocron/src/holocronModule.jsx]: line `80` was causing a ```react/destructuring-assignment``` eslint error to be raised and destructuring it to `loadCount` caused me to compare the same vars, decided to pass this a param to the `initiateLoad` function but this might have caused the behaviour of the function to change.

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 14 above)
Comment (nellyk) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: once the new version of eslint-config-amex is released i'll revert this changed

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 6 above)
Comment (nellyk) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: once new version of eslint-config-amex is released i shall revert this

File: packages/holocron/__tests__/loadModule.web.spec.js (hunk 17 above)
Comment (nellyk) [reply packages/holocron/__tests__/loadModule.web.spec.js]: is this what you meant? let me know if this works or if you have any other suggestions

File: packages/holocron/src/ducks/load.js (hunk 8 above)
Comment (nellyk) [reply packages/holocron/src/ducks/load.js]: reverted this changes in 56a4136

File: packages/holocron/src/loadModule.web.js (hunk 18 above)
Comment (nellyk) [reply packages/holocron/src/loadModule.web.js]: Seems to support everything above IE9 https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

File: packages/holocron/src/createHolocronStore.js [hunk 22]
```diff
@@ -85,41 +87,40 @@ function createReducer({ appReducer, state, localsForBuildInitialState }) {
...
-      ...extraThunkArguments,
-      rebuildReducer,
-      modules: store.modules,
-    })(middlewareAPI)(store.dispatch);
-
-    return {
-      ...store,
-      dispatch,
-    };
+const holocronEnhancer = (localsForBuildInitialState, extraThunkArguments = {}) => (
```
Comment (JAdshead) [inline packages/holocron/src/createHolocronStore.js]: if you have the arguments on a new line after `(` then the closing `)` should also be on a new line to help readability.

```js
const myFunction = (
  arg1,
  arg2
) => (
  // some code
);
```
or you can keep them on the same line.
```js
const myFunction = (arg1, arg2) => (
  // some code
);
```

File: packages/holocron/src/holocronModule.jsx [hunk 23]
```diff
@@ -55,9 +56,14 @@ export default function holocronModule({
         }
       }
 
-      componentWillReceiveProps(nextProps) {
-        if (shouldModuleReload && shouldModuleReload(this.props, nextProps)) {
-          const newLoadCount = this.state.loadCount + 1;
+      // ignoring to support deprecated componentWillReceiveProps.
+      // This needs to be removed to support React17
+      // eslint-disable-next-line camelcase
+      UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (JAdshead) [inline packages/holocron/src/holocronModule.jsx]: could this become `componentDidUpdate` instead ?

File: packages/holocron/src/updateModuleRegistry.js [hunk 24]
```diff
@@ -18,44 +18,44 @@ import loadModule from './loadModule.node';
...
   onModuleLoad = () => null,
-  batchModulesToUpdate = x => [x],
+  batchModulesToUpdate = (x) => [x],
   getModulesToUpdate = defaultGetModulesToUpdate,
 }) {
   const currentModuleMap = getModuleMap().toJS();
-  const modulesToUpdate =
-    batchModulesToUpdate(getModulesToUpdate(currentModuleMap.modules || {}, newModuleMap.modules));
+  const modulesToUpdate = batchModulesToUpdate(
+    getModulesToUpdate(currentModuleMap.modules || {}, newModuleMap.modules));
```
Comment (JAdshead) [inline packages/holocron/src/updateModuleRegistry.js]: this formatting pattern seems to be all over, are you running something like prettier on save ? if so i recommend disabling this rule.

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 15 above)
Comment (10xLaCroixDrinker) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: I find the separate lines easier to follow

File: packages/holocron/__tests__/loadModule.web.spec.js (hunk 16 above)
Comment (PixnBits) [reply packages/holocron/__tests__/loadModule.web.spec.js]: cb functions are stored in `map`, but are they accessed/used?

File: packages/holocron-module-route/src/ModuleRouteUtils.jsx [hunk 25]
```diff
@@ -17,22 +17,23 @@ import { createRoutes } from '@americanexpress/one-app-router';
...
-    [addToRouteProps(routes, newProps)];
+  return Array.isArray(routes) ? routes.map((route) => addToRouteProps(route, newProps))
+    : [addToRouteProps(routes, newProps)];
 };
 
-export const getRouteIndex = (routes, props) =>
-  (routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute);
+export const getRouteIndex = (routes, props) => (
+  routes && createRoutes(passChildrenProps(routes, props))[0].indexRoute
+);
```
Comment (PixnBits) [inline packages/holocron-module-route/src/ModuleRouteUtils.jsx]: no change required, but wanted to comment (positively 😅 PRs can be rough 😓) that I like this formatting

File: packages/holocron/src/updateModuleRegistry.js (hunk 24 above)
Comment (nellyk) [reply packages/holocron/src/updateModuleRegistry.js]: disabling this, would be reverting this changes

File: packages/holocron/__tests__/updateModuleRegistry.spec.js [hunk 26]
```diff
@@ -21,20 +21,18 @@ import {
 } from '../src/moduleRegistry';
 
 jest.mock('../src/loadModule.node', () => {
-  const createTimeoutPromise = (resolveWith, time) =>
-    new Promise(res => setTimeout(() => res(resolveWith), time));
-  return jest.fn((moduleName, moduleVersion) =>
-    createTimeoutPromise(() => `new ${moduleName}@${moduleVersion}`, 100));
+  const createTimeoutPromise = (
```
[the remark this answers is not in the document]
Comment (nellyk) [reply packages/holocron/__tests__/updateModuleRegistry.spec.js]: updated this let me know what you think 24de93f

File: packages/holocron/src/createHolocronStore.js (hunk 22 above)
Comment (nellyk) [reply packages/holocron/src/createHolocronStore.js]: Thank you Johnny i've updated this on 24de93f

File: packages/holocron/src/updateModuleRegistry.js (hunk 24 above)
Comment (JAdshead) [reply packages/holocron/src/updateModuleRegistry.js]: sorry, i meant in prettier

File: packages/holocron/src/holocronModule.jsx (hunk 23 above)
Comment (mtomcal) [reply packages/holocron/src/holocronModule.jsx]: Updating to `componentDidUpdate ` et al. should be a new effort to refactor Holocron to the proper lifecycle hooks when there's time to understand the implications of those changes.

File: packages/holocron/__tests__/loadModule.web.spec.js (hunk 16 above)
Comment (nellyk) [reply packages/holocron/__tests__/loadModule.web.spec.js]: yes i 'm using this to mock the cb functions within the test, not sure that answers the questions

File: packages/holocron-module-route/src/ModuleRouteUtils.jsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (10xLaCroixDrinker) [reply packages/holocron-module-route/src/ModuleRouteUtils.jsx]: @PixnBits FWIW, an equality check is more important than a greater than comparison

File: packages/holocron/src/updateModuleRegistry.js [hunk 27]
```diff
@@ -18,44 +18,47 @@ import loadModule from './loadModule.node';
...
     const nextModules = await Promise.all(moduleBatch.map(
-      moduleName => loadModule(moduleName, newModuleMap.modules[moduleName], onModuleLoad)
+      (moduleName) => loadModule(moduleName, newModuleMap.modules[moduleName], onModuleLoad)
     ));
     return [...loadedModules, ...nextModules];
   }, []);
-  updatedModules = updatedModules.reduce((acc, module, i) =>
-    ({ ...acc, [flatModulesToUpdate[i]]: module }), {});
+  updatedModules = updatedModules.reduce((
+    acc, module, i) => ({ ...acc, [flatModulesToUpdate[i]]: module }), {}
```
Comment (10xLaCroixDrinker) [inline packages/holocron/src/updateModuleRegistry.js]: weird line break here and below again

Comment (nellyk) [reply packages/holocron/src/updateModuleRegistry.js]: updated this f97a86a

File: packages/holocron/__tests__/loadModule.web.spec.js (hunk 17 above)
Comment (mtomcal) [reply packages/holocron/__tests__/loadModule.web.spec.js]: Also, should we be refactoring in this effort when the purpose is to update for linting rules? We should create separate refactor efforts if we wish to make fixes.
